server/src/main/java/org/opensearch/identity/ScheduledJobIdentityManager.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import org.opensearch.identity.tokens.AuthToken;
+
+/**
+ * An interface with methods used to provide security for scheduled jobs
+ *
+ * @opensearch.experimental
+ */
+public interface ScheduledJobIdentityManager {
+
+    /**
+     * Method implemented by an identity plugin to store user information for a scheduled job
+     * @param jobId The id of the scheduled job
+     * @param indexName The index where scheduled job details is stored
+     */
+    void saveUserDetails(String jobId, String indexName);
```

Comment:
This would be left up to an IdentityPlugin to implement, but the idea with the security plugin is that this user is the creator of the scheduled job. For the security plugin that does mean getting the authenticated user from the threadcontext and persisting it in an index that would contain all scheduled jobs and the user (owner) associated with them.

See https://github.com/opensearch-project/security/pull/2773 for an implementation in the Security plugin.